File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isValidBST(TreeNode* root) {
        if(!root) return true;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            TreeNode* node=q.front();
            q.pop();
            if(node->left && node->val<=node->left->val) return false;
            if(node->right && node->val>=node->right->val) return false;

            if(node->left) q.push(node->left);
            if(node->right) q.push(node->right);
        }
        return true;

    }
};

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    # res=[]
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        # res=[]
        if root is None:
            return []
        # if root.left is None and root.right is None:
        #     res.append(root.val)
        #     return
        l=self.inorderTraversal(root.left)
        # res.append(root.val)
        r=self.inorderTraversal(root.right)
        return l+[root.val]+r
        # def dfs(root):
        #     if root is None:
        #         return [] 
            # dfs(root.left)
            # res.append(root.val)
            # dfs(root.right)



        # dfs(root)   
        # return res

        